packages/components/src/SegmentedControl/SegmentedControl.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {
+  Children,
+  type PropsWithChildren,
+  useEffect,
+  useRef,
+} from "react";
+import styles from "./SegmentedControl.css";
+import { SegmentedControlProvider } from "./SegmentedControlProvider";
+import { SegmentedControlOption } from "./SegmentedControlOption";
+
+interface SegmentedControlProps<T> extends PropsWithChildren {
+  readonly selectedOption: T;
+  readonly onSelectOption: (view: T) => void;
+  readonly defaultOption: T;
+}
+
+export function SegmentedControl<T>({
+  onSelectOption,
+  defaultOption,
+  children,
+}: SegmentedControlProps<T>) {
+  const container = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    const length = Children.count(children);
+
+    if (container.current) {
+      container.current.style.setProperty(
+        "--segmentedControl--option-count",
```

Comment:
You could avoid the useEffect all together by leveraging data attributes to reflect the count.

```html
<div data-option-count={React.Children.count(children).length}
```

```css
.container {
  --segmentedCpntrol--option-count: attr(option-count, 1)
}
```

Reply:
ok with that approach, first I was getting `Property 'length' does not exist on type 'number'`.
That also that broke (I think) `grid-template-columns` because the options then stacked vertically.

<img width="955" alt="Screenshot 2024-10-23 at 2 31 00 PM" src="https://github.com/user-attachments/assets/a504c5ed-a6c4-4940-9e5e-097510b297f9">

What do you think about setting the variable in `style`. It works and gets rid of the `useEffect`

```
<div 
        style={
          {
            "--segmentedControl--option-count": Children.count(children),
          } as CSSProperties
        }
```

Reply:
Oh, that's purely due to my dumb brain forgetting that `.count(children)` returns you the number.

The attr check was then failing because the attribute lookup didn't contain anything and fell back to 1?

Reply:
Yeah that's my understanding.  I _think_ that setting this directly in the style prop is still ok and still helping us avoid re renders?